.github/workflows/ci.yml

```diff
@@ -116,18 +116,18 @@ jobs:
         with:
           go-version: '1.21'
 
-      - name: Import GPG key
-        id: import_gpg
-        uses: crazy-max/ghaction-import-gpg@ea88154188003ca5aeb616063b2d0dd6a9cf86e2
-        with:
-          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
-          passphrase: ${{ secrets.GPG_PASSPHRASE }}
-
       - name: Configure asdf and plugins needed
         uses: asdf-vm/actions/install@83133f03f5693901c2296a8e622955087dc20267
 
+
+      - name: Write signing key to disk
+        run: 'echo "$KEY" > cosign.key'
+        shell: bash
+        env:
+          KEY: ${{ secrets.COSIGN_PRIVATE_KEY }}
+      
```

Comment:
```suggestion
```

this is unsafe for at least 2 reasons:
1. `echo $KEY` is stored in the shell history of the CI runner
2. running the workflow in debug mode can show what each line is interpreting and then the KEY is leaked in the logging.

cosign supports a `env://ENV_NAME` option to read the key from the env.